src/main/java/org/craftercms/studio/impl/v1/service/GeneralLockServiceImpl.java

```diff
@@ -61,6 +63,20 @@ public void lock(@ValidateStringParam String objectId) {
         }
     }
 
+    @Override
+    public void lock(final String... lockKeys) {
+        logger.debug("Thread '{}' will attempt to lock multiple objects '{}'", Thread.currentThread().getName(), lockKeys);
+        Arrays.stream(lockKeys).sorted().forEach(this::lock);
```

Comment:
We are sorting the `lockKeys` to try to prevent the issue with ABCD - DEB.
It will wait until the resource is unlocked. Problem with making this method `synchronized` is that it will prevent other threads from acquiring any locks, even if unrelated.